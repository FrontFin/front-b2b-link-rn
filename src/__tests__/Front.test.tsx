/* eslint-disable */
import React from 'react';
import { render } from '@testing-library/react-native';
import { FrontFinance } from '../index';

jest.mock('react-native-webview', () => {
  const { View } = require('react-native');
  return {
    WebView: View,
  };
});

// Mock callback functions
const mockOnError = jest.fn();
const SAMPLE_CATALOG_URL = 'https://web.getfront.com:443/b2b-iframe/01xs1-test'
const SAMPLE_LINK_TOKEN = 'aHR0cHM6Ly93ZWIuZ2V0ZnJvbnQuY29tL2IyYi1pZnJhbWUvdGVzdC1hY2NvdW50LXJhbmRvbS9icm9rZXItY29ubmVjdC9jYXRhbG9nMQ=='
const SAMPLE_MESH_URL = 'https://web.meshconnect.com'

describe('FrontFinance Component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly when URL is provided', () => {
    const tree = render(<FrontFinance url={SAMPLE_CATALOG_URL} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles onError callback for invalid URL', () => {
    render(<FrontFinance url={'invalid'} onError={mockOnError} />);

    expect(mockOnError).toHaveBeenCalledWith('Invalid catalog link provided');
  });

  it('renders sdk if valid catalogLink is provided', () => {
    render(<FrontFinance url={SAMPLE_CATALOG_URL} />);
  });

  it('handles onError callback for invalid link token', () => {
    render(<FrontFinance linkToken={'invalid=='} onError={mockOnError} />);

    expect(mockOnError).toHaveBeenCalledWith('Invalid link token provided');
  });

  it('renders sdk if valid catalog link is provided', () => {
    render(<FrontFinance linkToken={SAMPLE_LINK_TOKEN} />);
  });
});
