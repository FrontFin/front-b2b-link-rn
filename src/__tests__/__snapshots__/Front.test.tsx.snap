// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontFinance Component renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
  testID="front-finance-component"
/>
`;

exports[`FrontFinance Component renders correctly when URL is provided 1`] = `
<RCTSafeAreaView
  style={
    {
      "flex": 1,
    }
  }
  testID="front-finance-component"
/>
`;
